pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/compression/GzipDecompressor.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.io.compression;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.zip.DataFormatException;
+import java.util.zip.Inflater;
+import org.apache.pinot.segment.spi.compression.ChunkDecompressor;
+
+
+/**
+ * Implementation of {@link ChunkDecompressor} using GZIP decompression algorithm.
+ */
+class GzipDecompressor implements ChunkDecompressor {
+
+  private final Inflater _decompressor;
+
+  public GzipDecompressor() {
+    _decompressor = new Inflater();
+  }
+
+  @Override
+  public int decompress(ByteBuffer compressedInput, ByteBuffer decompressedOutput)
+      throws IOException {
+    _decompressor.reset();
+    _decompressor.setInput(compressedInput);
+    try {
+      _decompressor.inflate(decompressedOutput);
+    } catch (DataFormatException e) {
+      throw new IOException(e);
+    }
+    decompressedOutput.flip();
+    return decompressedOutput.limit();
+  }
+
+  @Override
+  public int decompressedLength(ByteBuffer compressedInput) {
```

Comment:
I realized this problem after merging it.. Is there a way to get the decompressed size from the compressed buffer? We cannot really use this decompressor if it cannot return the size properly. For `LZ4`, we do have `LZ4_WITH_LENGTH` decompressor as a replacement

Reply:
ok, looking into it.

Reply:
well, according to https://www.rfc-editor.org/rfc/rfc1952 , the last field of a block is `ISIZE` (Input SIZE), which is the size of the original (uncompressed) input data modulo 2^32. So if the orig data was greater than 4Gb then this field is no good. At this point I am going to call this acceptable enough for my own standards and I am going to move on.

Reply:
@marregui I don't think we need to handle the case of chunk over 4GB (or 2GB for java int). This method is critical for raw index V4 to work (you may check the usage of this method), so we should at least return a positive value.
Currently the test doesn't fail because we still use raw index V2 as default, but we are going to move to V4 as default soon.

Reply:
I have created https://github.com/apache/pinot/pull/12802 to add the size; I have also added explicit tests and ensured nothing breaks.